xgboost_ray/tests/test_end_to_end.py

```diff
@@ -99,6 +103,43 @@ def testJointTraining(self):
         pred_y = bst.predict(x_mat)
         self.assertSequenceEqual(list(self.y), list(pred_y))
 
+    def testTrainPredict(self, init=True, remote=None):
+        """Train with evaluation and predict"""
+        if init:
+            ray.init(num_cpus=2, num_gpus=0)
+
+        dtrain = RayDMatrix(self.x, self.y)
+
+        evals_result = {}
+        bst = train(
+            self.params,
+            dtrain,
+            ray_params=RayParams(num_actors=2),
+            evals=[(dtrain, "dtrain")],
+            evals_result=evals_result,
+            _remote=remote)
+
+        self.assertTrue("dtrain" in evals_result)
+
+        x_mat = RayDMatrix(self.x)
+        pred_y = predict(bst, x_mat, _remote=remote)
+        self.assertSequenceEqual(list(self.y), list(pred_y))
+
+    def testTrainPredictRemote(self):
+        """Train with evaluation and predict in a remote call"""
+        self.testTrainPredict(init=True, remote=True)
+
+    def testTrainPredictClient(self):
+        """Train with evaluation and predict in a client session"""
+        from ray.util.client.ray_client_helpers import ray_start_client_server
+
+        ray.init(num_cpus=2, num_gpus=0)
+        self.assertFalse(ray.util.client.ray.is_connected())
+        with ray_start_client_server():
+            self.assertTrue(ray.util.client.ray.is_connected())
+
+            self.testTrainPredict(init=False, remote=None)
```

Comment:
Is there a way to confirm that _remote is being detected automatically when used with client?

Reply:
Hm I think the assertTrue should be enough. But I'll think about an easy way to check this